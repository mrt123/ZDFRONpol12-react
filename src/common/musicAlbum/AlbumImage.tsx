interface AlbumImageProps {
  imageUrl: string | undefined;
}

const getCoverImageStyle = (imageUrl: string | undefined) => {
  return {
    width: "100px",
    height: "100px",
    background: "blue",
    backgroundImage: imageUrl === undefined ? "none" : `url("${imageUrl}")`,
    backgroundSize: "contain",
  };
};

const AlbumImage = ({ imageUrl }: AlbumImageProps) => {
  // we use &nbsp; so that JSDOM serializer would not void close div tag
  // TODO: remove this and instead force JSDOM to close tag properly in test setup
  return <div style={getCoverImageStyle(imageUrl)}>&nbsp;</div>;
};

export default AlbumImage;
