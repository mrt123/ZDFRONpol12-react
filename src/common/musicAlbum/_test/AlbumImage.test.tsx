import { render } from "@testing-library/react";
import AlbumImage from "../AlbumImage";

describe("AlbumImage", () => {
  it("url = lalala", () => {
    const { container } = render(<AlbumImage imageUrl="lalala" />);
    expect(container).toMatchSnapshot();
  });

  it("url = undefined", () => {
    const { container } = render(<AlbumImage imageUrl={undefined} />);
    expect(container).toMatchSnapshot();
  });
});
