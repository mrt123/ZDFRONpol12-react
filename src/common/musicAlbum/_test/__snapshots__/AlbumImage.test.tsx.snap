// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumImage url = lalala 1`] = `
<div>
  <div
    style="width: 100px; height: 100px; background: blue; background-image: url(lalala); background-size: contain;"
  >
     
  </div>
</div>
`;

exports[`AlbumImage url = undefined 1`] = `
<div>
  <div
    style="width: 100px; height: 100px; background: blue; background-image: none; background-size: contain;"
  >
     
  </div>
</div>
`;
